File: java/core/src/main/java/com/github/fernthedev/lightchat/core/exceptions/DebugChainedException.java
package com.github.fernthedev.core.exceptions;

public class DebugChainedException extends DebugException {

    private Exception exception;
    private String message;

    public DebugChainedException() {
        new IllegalArgumentException("Use DebugException instead").printStackTrace();
    }

    public DebugChainedException(Exception exception,String message) {
        super(message);
        this.exception = exception;
    }

    @Override
    public void printStackTrace() {
        System.out.println(message);
        super.printStackTrace();
        initCause(exception);
        System.err.println("Caused by: " + getCause());
        getCause().printStackTrace();
    }
}

File: java/core/src/main/java/com/github/fernthedev/lightchat/core/exceptions/DebugChainedException.kt
package com.github.fernthedev.lightchat.core.exceptions

class DebugChainedException : DebugException {
    private var exception: Exception? = null
    override val message: String? = null

    constructor() {
        IllegalArgumentException("Use DebugException instead").printStackTrace()
    }

    constructor(exception: Exception?, message: String?) : super(message) {
        this.exception = exception
    }

    override fun printStackTrace() {
        println(message)
        super.printStackTrace()
        initCause(exception)
        System.err.println("Caused by: $cause")
        cause!!.printStackTrace()
    }
}